.github/workflows/ci.yml

```diff
@@ -227,6 +271,11 @@ jobs:
         with:
           distribution: 'zulu'
           java-version: 11
+      - name: Configure Problem Matchers
+        run: |
+          echo "::add-matcher::.github/problem-matcher.json"
+          # remove default Problem Matchers, since we're using the scacap/action-surefire-report at the end
+          echo "::remove-matcher owner=java::"
```

Comment:
can we abstain from removing matchers?
i am concern we may be losing something, or be losing something in the future

Reply:
I figured that anything that would get logged should also be in the Surefire report, and processing structured data should be more reliable then applying regexes on unstructured logs.

Currently, there's only one default for generic exceptions: https://github.com/actions/setup-java/blob/main/.github/java.json and it produces useless annotations like:
![image](https://user-images.githubusercontent.com/795177/144255708-dd71fe0e-2f81-4075-a22b-5fc9ce4a6d5c.png)

Reply:
that depends on what "default Problem Matchers" does.
for example, if it can annotate problems related to tests crashing (Eg OOM), based on job log output, this would be example of something that won't make into structured surefire report.

that's just an example. I am asking what "default Problem Matchers" does today. It may do A today, and it may do A, B and C tomorrow.

Reply:
You gave a good example. I wanted to get rid of annotations that are marked as errors but in successful jobs, which might be confusing. But I'll drop this for now and we can get back to this later, in another PR. I'll subscribe to releases of the `setup-java` action to know when any new matchers will be added.